src/price_data.py

```diff
@@ -267,33 +269,41 @@ def _get_price_bitpanda_pro(
         # if there were no trades in the requested time frame, the
         # returned data will be empty
         for t in timeframes:
-            end = utc_time
-            begin = utc_time - datetime.timedelta(minutes=t)
-
-            # https://github.com/python/mypy/issues/3176
-            params: dict[str, Union[int, str]] = {
-                "unit": "MINUTES",
-                "period": t,
-                "from": begin.isoformat(),
-                "to": end.isoformat(),
-            }
-            r = requests.get(baseurl, params=params)
-
-            assert r.status_code == 200
+            # if no trades can be found, move 30 min window to the past
+            for o in range(0, 300, 30):
```

Comment:
Shouldn't this be (0, 30, 60) ?
We might want to adjust this for easier understanding to
```
num_retries = 0
for num_retries in range(3):
    # If no trades can be found, move window each time 30 min to the past
    o = num_retries * 30
    ....
```